intel_adsp: Add a l1 cache init helper

ceolin opened the pull request:
l1 cache may vary, cavs25 and ace_v1x for example require power on
cache SRAM banks and configure between full or one way cache.

To keep this simple, lets add a helper function that is defined by each
target so they can do what they need individually.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

nashif commented on soc/xtensa/intel_adsp/common/include/cpu_init.h:
```diff
@@ -16,52 +17,7 @@ static ALWAYS_INLINE void cpu_early_init(void)
 {
 	uint32_t reg;
 
-#ifdef CONFIG_SOC_SERIES_INTEL_CAVS_V25
```
Why not just change this to be an SoC feature check via a Kconfig  instead of checking for specific SoCs and enable this for SoCs supporting this instead? 
This will avoid duplication of code and will keep the SoC initialization path in one place.
Also note that this is standard code flow for xtensa, see `arch/xtensa/core/startup/reset-vector.S` for example.

ceolin replied:
I thought about it, that is indeed a solution. Was just afraid that upcoming targets could have different initialization.
But, lets deal with the closes alligator. Will change it to use a Kconfig option.

andyross reviewed (commented):
Hits my cut/paste peeve again.  This is duplicating what amounts to xtensa-generic code (MEMCTL/PREFCTL and the RPO region setup are all LXn CPU features, really that could all move into the arch/xtensa tree and be driven via kconfig or whatever) so that we can add a little code before it on ACE?   Seems a little clumsy to me.

Commit:
arch: xtensa: Add option to control cache init

l1 cache may vary, cavs25 and ace_v1x for example require power on
cache SRAM banks.

Just add a hidden option to inform whether or not the target needs to
power on SRAM banks used for cache.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

kv2019i reviewed (approved):
Good with this, but this does now conflict with https://github.com/zephyrproject-rtos/zephyr/pull/48488/commits/7edf7c94588b3bf1d4c7aed5c33d1a03746849e4 of #48488 by @edersondisouza , which defines this via DTS.

andyross reviewed (approved):
Just noticed this got updated.  This version looks fine to me.  But we probably should look at the arch unification idea at some point in the near future.  There will be essentially identical Xtensa LXn core implementations arriving for AMD and Mediatek in the medium term future, and the NXP devices look very similar too.  Would be nice to get them all unified.

ceolin commented:
> Good with this, but this does now conflict with [7edf7c9](https://github.com/zephyrproject-rtos/zephyr/commit/7edf7c94588b3bf1d4c7aed5c33d1a03746849e4) of #48488 by @edersondisouza , which defines this via DTS.

yep, this pr is older, but that one has more changes, I'm fine with that implementation. We can move along with that pr IMHO
